[Test] Make bounding-rect-zoom.html use testharness.js

mdubet opened the pull request:
#### 340e2e865b87d707197f525eb20641d164d5f95f
<pre>
[Test] Make bounding-rect-zoom.html use testharness.js
<a href="https://bugs.webkit.org/show_bug.cgi?id=300998">https://bugs.webkit.org/show_bug.cgi?id=300998</a>

Reviewed by Tim Nguyen.

This test the old behavior of the API (unzoomed rect) which is what we currently implement.

* LayoutTests/fast/transforms/bounding-rect-zoom-expected.txt: Added.
* LayoutTests/fast/transforms/bounding-rect-zoom.html:
* LayoutTests/platform/glib/fast/transforms/bounding-rect-zoom-expected.txt: Removed.
* LayoutTests/platform/gtk/fast/transforms/bounding-rect-zoom-expected.png: Removed.
* LayoutTests/platform/ios/fast/transforms/bounding-rect-zoom-expected.png: Removed.
* LayoutTests/platform/ios/fast/transforms/bounding-rect-zoom-expected.txt: Removed.
* LayoutTests/platform/mac/fast/transforms/bounding-rect-zoom-expected.png: Removed.
* LayoutTests/platform/mac/fast/transforms/bounding-rect-zoom-expected.txt: Removed.
* LayoutTests/platform/win/fast/transforms/bounding-rect-zoom-expected.txt: Removed.

Canonical link: <a href="https://commits.webkit.org/301740@main">https://commits.webkit.org/301740@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/39c553da7cbdb157e68cc6bc8ae53b398542b517

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/126883 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/46523 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/37505 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/133887 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/78477 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/3464c598-0cba-4966-8b74-d4308f40190c) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47146 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55054 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/96558 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/78477 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/2219d95f-c67c-46a3-8726-1151d906a893) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/129831 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37725 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/113490 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77072 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/f6194984-e0c2-441f-ac02-658682d330a6) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/36591 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/31665 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77281 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/107559 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31975 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136413 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53549 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/41231 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/105068 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54049 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/109852 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104764 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26719 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50277 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/28612 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/51025 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53479 "Built successfully") | | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/52728 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/56063 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54480 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/fast/transforms/bounding-rect-zoom.html:
```diff
@@ -1,75 +1,41 @@
+    assert_approx_equals(bounds.right, 200, 0.01, 'right should be scaled by CSS zoom');
+    assert_approx_equals(bounds.bottom, 200, 0.01, 'bottom should be scaled by CSS zoom');
+    assert_approx_equals(bounds.width, 100, 0.01, 'width should be scaled by CSS zoom');
+    assert_approx_equals(bounds.height, 100, 0.01, 'height should be scaled by CSS zoom');
```
I think you mean `should not be`, at least with the current expectations

nt1m commented on LayoutTests/fast/transforms/bounding-rect-zoom.html:
```diff
@@ -1,75 +1,41 @@
+    assert_approx_equals(rects[0].right, 200, 0.01, 'right should be scaled by CSS zoom');
+    assert_approx_equals(rects[0].bottom, 200, 0.01, 'bottom should be scaled by CSS zoom');
+    assert_approx_equals(rects[0].width, 100, 0.01, 'width should be scaled by CSS zoom');
+    assert_approx_equals(rects[0].height, 100, 0.01, 'height should be scaled by CSS zoom');
```
ditto here

nt1m reviewed (approved):
I would use this title:

Make bounding-rect-zoom use testharness.js

mdubet committed:
[Test] Make bounding-rect-zoom.html use testharness.js
https://bugs.webkit.org/show_bug.cgi?id=300998

Reviewed by Tim Nguyen.

This test the old behavior of the API (unzoomed rect) which is what we currently implement.

* LayoutTests/fast/transforms/bounding-rect-zoom-expected.txt: Added.
* LayoutTests/fast/transforms/bounding-rect-zoom.html:
* LayoutTests/platform/glib/fast/transforms/bounding-rect-zoom-expected.txt: Removed.
* LayoutTests/platform/gtk/fast/transforms/bounding-rect-zoom-expected.png: Removed.
* LayoutTests/platform/ios/fast/transforms/bounding-rect-zoom-expected.png: Removed.
* LayoutTests/platform/ios/fast/transforms/bounding-rect-zoom-expected.txt: Removed.
* LayoutTests/platform/mac/fast/transforms/bounding-rect-zoom-expected.png: Removed.
* LayoutTests/platform/mac/fast/transforms/bounding-rect-zoom-expected.txt: Removed.
* LayoutTests/platform/win/fast/transforms/bounding-rect-zoom-expected.txt: Removed.

Canonical link: https://commits.webkit.org/301740@main
